integration_tests/src/main/python/json_fuzzing_test.py

```diff
@@ -0,0 +1,413 @@
+    path = random.randint(0, 2)
+    if path == 0:
+        for t in gen_digit():
+            yield t
+    elif path == 1:
+        yield chr(random.randint(0x41, 0x46))
+    else:
+        yield chr(random.randint(0x61, 0x66))
+
+def gen_number():
+    """
+    NUMBER -> INTEGER FRACTION EXPONENT
+    """
+    for t in gen_integer():
+        yield t 
+    for t in gen_fraction():
+        yield t 
+    for t in gen_exponent():
+        yield t 
+
+def gen_integer():
+    """
+    INTEGER -> DIGIT 
+             | ONENINE DIGITS 
+             | '-' DIGIT 
+             | '-' ONENINE DIGITS
+    """
+    if random.randint(1, 100) <= 50:
+        yield '-'
+    
+    if random.randint(1, 100) <= 50:
+        for t in gen_digit():
+            yield t 
+    else:
+        for t in gen_onenine():
+            yield t 
+        for t in gen_digits():
+            yield t 
+
+def gen_digits():
+    """
+    DIGITS -> DIGIT 
+            | DIGIT DIGITS
+    """
+    for t in gen_digit():
+        yield t 
+
+    if random.randint(1, 100) < 70:
+        for t in gen_digits():
+            yield t 
+
+def gen_digit():
+    """
+    DIGIT -> '0' 
+           | ONENINE
+    """
+    if random.randint(0, 9) == 0:
+        yield '0'
+    else:
+        for t in gen_onenine():
+            yield t
+
+def gen_onenine():
+    """
+    ONENINE -> '1' .. '9'
+    """
+    yield chr(random.randint(0x31, 0x39))
+
+def gen_fraction():
+    """
+    FRACTION -> "" | '.' DIGITS
+    """
+    if random.randint(1, 100) < 50:
+        yield ""
+    else:
+        yield '.'
+        for t in gen_digits():
+            yield t 
+
+def gen_exponent():
+    """
+    EXPONENT -> "" 
+              | 'E' SIGN DIGITS 
+              | 'e' SIGN DIGITS
+    """
+    if random.randint(1, 100) < 20:
+        yield ""
+    else:
+        yield random.choice(['E', 'e'])
+        for t in gen_sign():
+            yield t 
+        for t in gen_digits():
+            yield t 
+
+def gen_sign():
+    """
+    SIGN -> "" 
+          | '+' 
+          | '-'
+    """
+    yield random.choice(["", '+', '-'])
+
+def gen_whitespace():
+    """
+    WHITESPACE -> '' 
+                | 0x0020 WHITESPACE 
+                | 0x000a WHITESPACE (todo)
+                | 0x000d WHITESPACE (todo)
+                | 0x0009 WHITESPACE (todo)
+    """
+    if random.randint(0, 4) > 3:
+        yield chr(random.choice([0x0020]))
+        for t in gen_whitespace():
+            yield t
+    else:
+        yield ''
+
+def gen_bool():
+    """
+    BOOL -> "true" 
+          | "null" 
+          | "false"
+    """
+    yield random.choice(["true", "null", "false"])
+
+
+
+_enable_all_types_conf = {
+    'spark.rapids.sql.format.json.enabled': 'true',
+    'spark.rapids.sql.format.json.read.enabled': 'true'}
+
+@approximate_float
+@allow_non_gpu('FileSourceScanExec')
+@pytest.mark.xfail(reason="fuzz test may randomly fail")
+def test_json_read_fuzz(spark_tmp_path):
+    depth = random.randint(1, 5)
```

Comment:
How do we reproduce/debug an error when it happens? Everything is random and based off of a regular random. Typically in python we would want to create an instance of Random with a seed, and then pass it everywhere so the results can be reproduced. We do this in data_gen already.

https://github.com/NVIDIA/spark-rapids/blob/00630537365c034186fa9031694edb503e4341a6/integration_tests/src/main/python/data_gen.py#L671

I am fine if we randomly generate a seed based off of a timestamp or something like that. But at a minimum we need a way to log what that seed was so if/when it does fail we can debug things.  With xdist this is not super simple, so I would suggest that we catch all Exceptions and then wrap them with something new that includes the seed that triggered the problem.

Reply:
Add the flag `--debug_tmp_path` can save the test data